ligo/src/types.mligo

```diff
@@ -21,6 +21,23 @@ type ledger_key = address * token_id
 type ledger_value = nat
 type ledger = (ledger_key, ledger_value) big_map
 
+type freeze_history_entry =
+  { period : nat
+  ; staked : nat
+  ; unstaked : nat
+  }
+
+type freeze_history = (address, freeze_history_entry list) big_map
```

Comment:
My additional storage space improvement is this:
we actually no longer need `freeze_history_entry list` now that we have `compact_freeze_history`, as that can contain all the necessary informations.

So my suggestion is to:
1. rename `compact_freeze_history` to `address_freeze_history`
2. change `freeze_history` to `type freeze_history = (address, address_freeze_history) big_map`
3. deal with `address_freeze_history` directly in the code, removing the conversions from and to the `freeze_history_entry list` that we currently have.

Does that sound reasonable to you?